### PR TITLE
feat: dpmodel energy loss & consistent tests

### LINKED ISSUE #4105: Add universal/consistent tests for loss modules
> > > > ```diff
> > > > + Hits        42870    43087     +217     
> > > > - Misses       7822     7934     +112     
> > > > - Partials      946      948       +2     
> > > > ```
> > > 
> > > 
> > > I see that 1/3 of the new codes have not been tested.
> > 
> > 
> > Where can I see these datas about missing UT? I will add UT based on these datas.
> 
> You can click the link sent by @codecov or click the codecov checks.
> 
> It looks like the loss module is not tested. @iProzd Do we have a universal test fixture for loss functions?

Not yet, maybe we need a discussion to design a universal test for loss modules.

_Originally posted by @iProzd in https://github.com/deepmodeling/deepmd-kit/issues/3867#issuecomment-2331399624_

### LINKED ISSUE #4429: [Feature Request] Array API loss
### Summary

Migrate PT losses to Array API losses.

### Detailed Description

Need the consistent tests for loss.

### Further Information, Files, and Links

_No response_

### PR DESCRIPTION
**Pull request by njzjz**
Fix #4105. Fix #4429.

## Release Notes

- **New Features**
  - Introduced a new energy loss calculation framework with support for multiple machine learning backends.
  - Added serialization and deserialization capabilities for loss modules.
  - Added a new class `EnergyLoss` for computing energy-related loss metrics.
  
- **Documentation**
  - Added SPDX license identifiers to multiple files.
  - Included docstrings for new classes and methods.

- **Tests**
  - Implemented comprehensive test suite for energy loss functions across different platforms (TensorFlow, PyTorch, Paddle, JAX).
  - Introduced a new test class `TestEner` for evaluating energy loss functions.

### CODE REVIEW

File: deepmd/dpmodel/loss/ener.py [hunk 1]
```diff
@@ -0,0 +1,337 @@
...
+                drdq, [-1, natoms[0] * 3, self.numb_generalized_coord]
+            )
+            gen_force_hat = xp.einsum(
+                "bij,bi->bj", drdq_reshape, force_hat_reshape_nframes
+            )
+            gen_force = xp.einsum("bij,bi->bj", drdq_reshape, force_reshape_nframes)
+            diff_gen_force = gen_force_hat - gen_force
+            l2_gen_force_loss = xp.mean(xp.square(diff_gen_force))
+            pref_gf = find_drdq * (
+                self.limit_pref_gff
```
[the remark this answers is not in the document]
Comment (njzjz) [reply deepmd/dpmodel/loss/ener.py]: Fixed.

**Review by wanghan-iapcm — APPROVED** [review]

**Review by iProzd — APPROVED** [review]
